src/main/kotlin/com/exactpro/th2/read/file/common/AbstractFileReader.kt

```diff
@@ -202,6 +204,13 @@ abstract class AbstractFileReader<T : AutoCloseable>(
                     }
 
                     finalContent.also { content ->
+                        if (content.size == 1 && lastState == FileHolder.ReadState.START && fileHolder.readState == FileHolder.ReadState.FIN) {
+                            content.first().markSingle()
+                        } else {
+                            if (lastState == FileHolder.ReadState.START) content.first().markFirst()
+                            if (fileHolder.readState == FileHolder.ReadState.FIN) content.last().markLast()
+                        }
```

Comment:
I think it looks better as a `when` statement
```suggestion
                        when {
                          content.size == 1 && lastState == FileHolder.ReadState.START && fileHolder.readState == FileHolder.ReadState.FIN ->
                            content.first().markSingle()
                          lastState == FileHolder.ReadState.START -> content.first().markFirst()
                          fileHolder.readState == FileHolder.ReadState.FIN -> content.last().markLast()
                        }
```

Reply:
In your example chain when two statments can be true isnt allowed, one group can have first and also last states

Reply:
Yes, you are right. My mistake